hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java

```diff
@@ -622,13 +597,24 @@ public void handle(Packet packet) {
         }
     }
 
+    @Override
+    public Operation getPreJoinOperation() {
+        // pre-join operations are only sent by master member
+        return getRegistrationOperation();
+    }
+
+    @Override
+    public Operation getPostJoinOperation() {
+        // send post join operation only if this is the newly joining member
+        return nodeEngine.getClusterService().isMaster() ? null : getRegistrationOperation();
```

Comment:
While operating in 3.8 cluster version, shouldn't the post-join op be delivered unconditionally?

Reply:
you're right. I forgot 3.8 compat here completely.